internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -179,6 +179,12 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 			pluginsdk.ForceNewIfChange("cross_region_restore_enabled", func(ctx context.Context, old, new, meta interface{}) bool {
 				return old.(bool) && !new.(bool)
 			}),
+			pluginsdk.ForceNewIfChange("immutability", func(ctx context.Context, old, new, meta interface{}) bool {
+				if old.(string) != new.(string) && old.(string) == string(vaults.ImmutabilityStateLocked) {
```

Comment:
Is there a case that `old != new`?, also this function can be simplified to a single `return` expression?

Reply:
I'm not sure when will this func be called, worried if it will be always called ( it will only be called when there is diff dectected? checked code but not 100% sure ).